### PR TITLE
Ineffassign fixes for pkg/volume

### PR DESCRIPTION
**Pull request by alena1108**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Some places in call had ineffectual assignments. This PR addresses them in `pkg/volume`

**Which issue(s) this PR fixes**:

Fixes #

N/A

**Special notes for your reviewer**:

This PR is an offspring of https://github.com/kubernetes/kubernetes/pull/87462 where it was suggested to split the fixes on per package basis

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

cc @lavalamp

### CODE REVIEW

File: pkg/volume/azure_file/azure_file.go [hunk 1]
```diff
@@ -114,6 +114,11 @@ func (plugin *azureFilePlugin) newMounterInternal(spec *volume.Spec, pod *v1.Pod
 		return nil, err
 	}
 	secretName, secretNamespace, err := getSecretNameAndNamespace(spec, pod.Namespace)
+	if err != nil {
+		// Log-and-continue instead of returning an error for now
+		// due to unspecified backwards compatibility concerns (a subject to revise)
+		klog.V(4).Infof("Failed to get secret name and namespace: %v", err)
```
Comment (andyzhangx) [inline pkg/volume/azure_file/azure_file.go]:
```
klog.Errorf("get secret name and namespace from pod(%s) return with error: %v", pod.Name, err)
```

File: pkg/volume/util/fsquota/project.go [hunk 2]
```diff
@@ -324,7 +324,7 @@ func createProjectID(path string, ID common.QuotaID) (common.QuotaID, error) {
 	if err == nil {
 		defer closeProjectFiles(fProjects, fProjid)
 		list := readProjectFiles(fProjects, fProjid)
-		writeProjid := true
+		var writeProjid bool
```
Comment (andyzhangx) [inline pkg/volume/util/fsquota/project.go]: I think default value of bool is false

File: pkg/volume/util/fsquota/project.go [hunk 3]
```diff
@@ -345,7 +345,7 @@ func removeProjectID(path string, ID common.QuotaID) error {
...
```
Comment (andyzhangx) [inline pkg/volume/util/fsquota/project.go]: I think default value of bool is false

File: pkg/volume/util/fsquota/project.go (hunk 2 above)
Comment (alena1108) [reply pkg/volume/util/fsquota/project.go]: @anderseknert that's correct, but the default value doesn't have any affect here as the `writeProjid` value is being reset on the next line anyway

File: pkg/volume/util/fsquota/project.go (hunk 3 above)
Comment (alena1108) [reply pkg/volume/util/fsquota/project.go]: @anderseknert that's correct, but the default value doesn't have any affect here as the `writeProjid` value is being reset on the next line anyway.

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
for azure part

**Comment by anderseknert** [comment]
For a while I felt important being called in to provide advice on a topic I know nothing of, but I'm pretty sure I'm tagged by accident here, @alena1108 :)

**Pull request merged**